src/generators/testing/TestingGenerator.res

```diff
@@ -98,6 +98,76 @@ let drawGrid = () => {
   }
 }
 
+let drawSun = (ox, oy, ~color, ~width) => {
+  let angles = [
+    0.0,
+    15.0,
+    30.0,
+    45.0,
+    60.0,
+    75.0,
+    90.0,
+    105.0,
+    120.0,
+    135.0,
+    150.0,
+    165.0,
+    180.0,
+    195.0,
+    210.0,
+    225.0,
+    240.0,
+    255.0,
+    270.0,
+    285.0,
+    300.0,
+    315.0,
+    330.0,
+    345.0,
+  ]
+  let innerRadius = 20.0
+  let outerRadius = 50.0
+  angles->Belt.Array.forEach(angle => {
+    let radians = angle *. Js.Math._PI /. 180.0
+    let x1 = Belt.Float.toInt(innerRadius *. Js.Math.cos(radians))
+    let y1 = Belt.Float.toInt(innerRadius *. Js.Math.sin(radians))
+    let x2 = Belt.Float.toInt(outerRadius *. Js.Math.cos(radians))
+    let y2 = Belt.Float.toInt(outerRadius *. Js.Math.sin(radians))
+    Generator.drawLine((ox + x1, oy + y1), (ox + x2, oy + y2), ~color, ~width, ())
+  })
+}
+
+let drawSquare = (ox, oy, ~color, ~width) => {
+  let size = 50
+  let ox = ox - size / 2
+  let oy = oy - size / 2
+  let p1 = (ox, oy)
+  let p2 = (ox + size, oy)
+  let p3 = (ox + size, oy + size)
+  let p4 = (ox, oy + size)
+  Generator.drawLine(p1, p2, ~color, ~width, ())
+  Generator.drawLine(p2, p3, ~color, ~width, ())
+  Generator.drawLine(p3, p4, ~color, ~width, ())
+  Generator.drawLine(p4, p1, ~color, ~width, ())
+}
+
+let drawLinesTestPage = () => {
+  Generator.usePage("Lines")
+  drawSun(100, 100, ~color="#000000", ~width=1.0)
+  drawSun(250, 100, ~color="#000000", ~width=2.0)
+  drawSun(400, 100, ~color="#000000", ~width=3.0)
+  drawSun(100, 250, ~color="#ff0000", ~width=1.0)
+  drawSun(250, 250, ~color="#00ff00", ~width=2.0)
+  drawSun(400, 250, ~color="#0000ff", ~width=3.0)
+
+  drawSquare(100, 400, ~color="#000000", ~width=1.0)
+  drawSquare(250, 400, ~color="#000000", ~width=2.0)
+  drawSquare(400, 400, ~color="#000000", ~width=3.0)
+  drawSquare(100, 550, ~color="#ff0000", ~width=1.0)
+  drawSquare(250, 550, ~color="#00ff00", ~width=2.0)
+  drawSquare(400, 550, ~color="#0000ff", ~width=3.0)
+}
```

Comment:
Let's add a test page for the fold lines as well. Here's an example:

```
let drawFoldBox = (x, y, size) => {
  let p1 = (x, y)
  let p2 = (x + size, y)
  let p3 = (x + size, y + size)
  let p4 = (x, y + size)
  Generator.drawFold(p1, p2)
  Generator.drawFold(p2, p3)
  Generator.drawFold(p3, p4)
  Generator.drawFold(p4, p1)
}

let drawFoldLinesTestPage = () => {
  Generator.usePage("Fold Lines")

  // Clockwise
  let ox = 40
  let oy = 70
  let size = 64
  Generator.drawText("Clockwise", (ox, oy - 20), 16)
  drawSteveHead("Steve", ox, oy, size)
  drawFoldBox(ox + size, oy + size, size)

  // Anti-clockwise
  let ox = 300
  let oy = 70
  let size = 64
  Generator.drawText("Anti-Clockwise", (ox, oy - 20), 16)
  drawSteveHead("Steve", ox, oy, size)
  drawFoldBox(ox + size, oy + size, size)

  Generator.drawText("Arbitrary Sizes", (40, 320), 16)

  let ox = 40
  let oy = 340

  for index in 0 to 19 {
    let size = 4 + index
    let x = ox + index * 27
    let y = oy
    drawFoldBox(x, y, size)
  }

  let ox = 40
  let oy = 380

  for index in 0 to 12 {
    let size = 24 + index
    let x = ox + index * 40
    let y = oy
    drawFoldBox(x, y, size)
  }

  Generator.drawText("Power of 2 Sizes", (40, 440), 16)

  let ox = 40
  let oy = 460

  for index in 0 to 4 {
    let exp = Belt.Int.toFloat(index) +. 2.0
    let size = Js.Math.pow_float(~base=2.0, ~exp)->Belt.Float.toInt
    let x = ox + index * 50
    let y = oy
    drawFoldBox(x, y, size)
  }

  Generator.drawText("Contrast", (40, 560), 16)
  let ox = 40
  let oy = 580
  let size = 128
  Generator.drawTexture("TextureColors64x64", (0, 0, 64, 64), (ox, oy, 128, 128), ())
  drawFoldBox(ox + 16, oy + 16, size - 32)
  drawFoldBox(ox + 48, oy + 48, size - 96)
}
```

Reply:
This drawFoldBox() function looks very similar to the drawRect() function I made earlier, except that drawRect() is the same whether it is done clockwise or anticlockwise. Since the folds functions' uses are all done in conjunction with that function and the ones related to it, I think that a testing page for the folds should be done when the other functions begin to be tested.

Reply:
That `drawFoldBox()` is not only intended as a "box" drawing function for folds, it's intended to also test what drawing individual fold lines look like when drawn in one direction and then the other direction.

If we add a `drawRect()` in future that also supports fold patterns, we should write new tests for that, and leave these existing ones unchanged.

Feel free to just copy/paste that code I wrote above into a new testing page. This has value for me so I can quickly see if there are visual regressions due to changes made in the future, thanks!